tee-worker/sidechain/rpc-handler/src/direct_top_pool_api.rs

```diff
@@ -333,27 +341,94 @@ where
 	response.map_err(|e| format!("{:?}", e))
 }
 
-async fn request_vc_inner(params: Params) -> Result<RpcReturnValue, String> {
+fn author_submit_request_vc_inner<R, TCS, G>(author: Arc<R>, params: Params) -> Result<H256, String>
+where
+	R: AuthorApi<H256, H256, TCS, G> + Send + Sync + 'static,
+	TCS: PartialEq + Encode + Decode + Debug + Send + Sync + 'static,
+	G: PartialEq + Encode + Decode + Debug + Send + Sync + 'static,
+{
 	let payload = get_request_payload(params)?;
-	let request = AesRequest::from_hex(&payload)
-		.map_err(|e| format!("AesRequest construction error: {:?}", e))?;
+	let request = AesRequest::from_hex(&payload).map_err(|e| format!("{:?}", e))?;
+
+	// Use this hash to trigger the storage of connection_registry
+	let hash = request.using_encoded(|x| BlakeTwo256::hash(x));
 
+	thread::spawn(move || request_vc_inner(author, request, hash));
```

Comment:
This does not look right, We are creating a thread every time we receive a request?

Reply:
Yes. Because `request_vc_inner` is going to process the request and wait for the response.

Reply:
I think what Faisal means is it could easily spawn many concurrent threads - it could cause problems especially for enclave where the number of threads are limited

Reply:
I think it should directly post the task to thread pool and return

Reply:
I think it depends on how jsonrpc handles the handling of sync method internally - the while loop might need to be executed on threads in the first threadpool

Reply:
This is the major change we need before we can merge it

Reply:
Thanks. I see the problem. @felixfaisal good catch.